plugins/todo-backend/src/lib/TodoReader/TodoScmReader.test.ts

```diff
@@ -189,4 +189,100 @@ describe('TodoScmReader', () => {
       'Failed to parse TODO in https://github.com/o/r/catalog-info.yaml at my-file.sh, Error: failed to parse',
     );
   });
+
+  it('should not filter out exclude folders', async () => {
+    const reader = mockReader();
+    const filePathFilter = jest.fn(() => true);
+
+    const todoReader = new TodoScmReader({
+      logger: getVoidLogger(),
+      reader,
+      integrations: ScmIntegrations.fromConfig(new ConfigReader({})),
+      filePathFilter,
+    });
+    reader.readTree.mockResolvedValueOnce({
+      files: async () => [
+        {
+          content: async () => Buffer.from('// TODO: my-todo', 'utf8'),
+          path: 'my-folder/my-file.js',
+        },
+      ],
+    } as ReadTreeResponse);
+    await expect(
+      todoReader.readTodos({
+        url: 'https://github.com/backstage/backstage/catalog-info.yaml',
+      }),
+    ).resolves.toEqual({
+      items: [
+        {
+          text: 'my-todo',
+          tag: 'TODO',
+          lineNumber: 1,
+          repoFilePath: 'my-folder/my-file.js',
+          viewUrl:
+            'https://github.com/backstage/backstage/my-folder/my-file.js#L1',
+        },
+      ],
+    });
+    expect(reader.readTree).toHaveBeenCalledTimes(1);
+    expect(reader.readTree).toHaveBeenCalledWith(
+      'https://github.com/backstage/backstage/catalog-info.yaml',
+      {
+        etag: undefined,
+        filter: expect.any(Function),
+      },
+    );
+    // Filter function should filter out exclude folders
+    const filterFunc = reader.readTree.mock.calls[0][1]!.filter!;
+    expect(filterFunc('my-file.js')).toBe(true);
+    expect(filterFunc('my-folder/my-file.js')).toBe(true);
+  });
+
+  it('should  filter out exclude folders', async () => {
+    const reader = mockReader();
+    const filePathFilter = jest.fn(() => false);
+
+    const todoReader = new TodoScmReader({
+      logger: getVoidLogger(),
+      reader,
+      integrations: ScmIntegrations.fromConfig(new ConfigReader({})),
+      filePathFilter,
+    });
+    reader.readTree.mockResolvedValueOnce({
+      files: async () => [
```

Comment:
Let's make this mock return an empty array, the test is a bit confusing otherwise 😅

Reply:
Sorry to confuse you 😅

Reply:
Hi! I see this one still isn't an empty array. After that I think this is good to go!

Reply:
Oh wait, am I misunderstanding the test perhaps - it's actually supposed to read this and then ignore the file 🤔 OK never mind then